src/gui/states.py

```diff
@@ -96,70 +90,42 @@ class WaitingState(State):
-
-    def tag_reader_timer_expired(self):
-        try:
-            if self.application.key_reader.tag_was_read():
-                self.tag_reader_timer_cancel()
-                tag_id = self.application.key_reader.get_aptus_tag_id()
-                self.application.on_event(Event(Event.TAG_READ, tag_id))
-                if self.application.state != self:
-                    return
-        except serial.serialutil.SerialException:
-            logger.exception("Key reader disconnected")
-            self.key_reader.close()
-            self.tag_reader_timer_cancel()
-            self.application.key_reader.com.close()
-            self.application.on_event(Event(Event.SERIAL_PORT_DISCONNECTED))
-            return
-        except KeyReaderNeedsRebootError:
-            self.application.on_event(Event(Event.SERIAL_PORT_DISCONNECTED))
-        except Exception:
-            logger.exception("Exception while polling key reader")
-
-        self.tag_reader_timer_start()
-
-    def tag_reader_timer_start(self):
-        self.tag_reader_timer = self.master.after(SERIAL_POLL_TIME_MS, self.tag_reader_timer_expired)
-
-    def tag_reader_timer_cancel(self):
-        self.master.after_cancel(self.tag_reader_timer)
+        if event == GuiEvent.LOGIN:
+            login = namedtuple("member_number", "pin_code")
+            login.member_number, login.pin_code = data
+            logger.debug(f"Login requested with member_numer = {login.member_number}, pin_code = {login.pin_code}")
+            self.application.on_event(Event(Event.LOGIN, login))
+        elif event == GuiEvent.PIN_CODE_REQUESTED:
+            member_number = data
+            self.application.on_event(Event(Event.PIN_CODE_REQUESTED, member_number))
 
     def on_event(self, event):
         super().on_event(event)
 
         event_type = event.event
-
-        if event_type == Event.TAG_READ:
+        if event_type == Event.LOGIN:
             self.gui.start_progress_bar()
-            self.gui.set_tag_status("Tag read...")
             self.master.update()
 
             try:
-                tagid = event.data
-                self.member = Member.from_tagid(self.application.makeradmin_client, tagid)
-                return MemberIdentified(self.application, self.master, self.member)
-            except NoMatchingTagId:
+                login = event.data
+                member = Member.from_member_number(self.application.makeradmin_client, login.member_number)
+                logger.debug(f"Login requested with member_numer = {login.member_number}, pin_code = {login.pin_code} - implement here")
+                return MemberIdentified(self.application, self.master, member)
+            except NoMatchingMemberNumber:
                 self.gui.reset_gui()
-                self.gui.show_error_message("Could not find a member that matches the specific tag")
+                self.gui.show_error_message("Login incorrect.")
```

Comment:
I don't think this exception can be thrown here. Either the login works or we get and `IncorrectPinCode` exception. The backend never tells the memberbooth why it did not work.